crates/task-impls/src/consensus/mod.rs

```diff
@@ -315,17 +308,19 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> ConsensusTaskState<TYPES, I>
         false
     }
 
+    /// Dummy proposal
     #[cfg(feature = "dependency-tasks")]
+    #[allow(clippy::unused_async, clippy::no_effect_underscore_binding)]
```

Comment:
Nit: can we remove this `publish_proposal` so that we know we'll propose whenever we call `publish_proposal`? Similarly, we have only one `vote_if_able`.

Reply:
I've sent an update to remove this.